api/src/model/request/play_asset_pack_state_request.gd

```diff
@@ -0,0 +1,99 @@
+# ##############################################################################
+#
+#	Copyright 2020 Google LLC
+#
+#	Licensed under the Apache License, Version 2.0 (the "License");
+#	you may not use this file except in compliance with the License.
+#	You may obtain a copy of the License at
+#
+#		https://www.apache.org/licenses/LICENSE-2.0
+#
+#	Unless required by applicable law or agreed to in writing, software
+#	distributed under the License is distributed on an "AS IS" BASIS,
+#	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#	See the License for the specific language governing permissions and
+#	limitations under the License.
+#
+# ##############################################################################
+#
+# Request object that handles asynchronous logic related to 
+# get_asset_pack_state() and provides most updated PlayAssetPackState of a
+# given asset pack.
+# 
+# This object provides relevant getters so that it is possible to retrieve
+# the updated states from this object once the request completes.
+#
+# ##############################################################################
+class_name PlayAssetPackStateRequest
+extends PlayAssetDeliveryRequest
+
+signal request_completed(did_succeed, pack_name, result, exception)
```

Comment:
I decided to add this pack_name argument since devs might use the same callback functions for multiple PlayAssetPackStateRequests.  (In that case if we ran into an exception devs might be unable to know which PlayAssetPackStateRequest caused it since request will be null)

Reply:
Yeah that is also another viable design option. I'm more in favor of the current design since it can make the signal's interface more self-explanatory. 
Having signals to emit self can also be unideal for ```state_updated``` signal in later stage, since this request object can be mutable after this signal (may need to duplicate in this case).

Reply:
You're right that the current approach is a bit more descriptive, but I think the suggested approach could simplify the test code and user code enough to be worth the tradeoff. For example they could use the same code for the yield case and the signal subscription casel. i.e. they would access the same properties. 

Right now there's a fair bit of duplication having to have every field in the request mirrored in the signal arguments.

You do have a good point about the object maybe getting changed. Would that be an issue in practice? It seems like this class is fairly immutable, since the only way to set values on it is through private methods.

Reply:
To summarize our discussion offline:
1. The current approach is closer to how Godot's [HTTPRequest](https://docs.godotengine.org/en/stable/classes/class_httprequest.html#class-httprequest) class handles signal arguments.
2. The request returned by on_complete will never change, but the request for state_update may be changed out of band (i.e. in another thread).
3. Users are unlikely to use both yield and signal subscription so having them use the same code path isn't strictly necessary.